torchvision/tv_tensors/_keypoints.py

```diff
@@ -0,0 +1,94 @@
+    """
+
+    canvas_size: Tuple[int, int]
+
+    def __new__(
+        cls,
+        data: Any,
+        *,
+        canvas_size: Tuple[int, int],
+        dtype: Optional[torch.dtype] = None,
+        device: Optional[Union[torch.device, str, int]] = None,
+        requires_grad: Optional[bool] = None,
+    ):
+        tensor: torch.Tensor = cls._to_tensor(data, dtype=dtype, device=device, requires_grad=requires_grad)
+        if tensor.ndim == 1:
+            tensor = tensor.unsqueeze(0)
+        elif tensor.shape[-1] != 2:
+            raise ValueError(f"Expected a tensor of shape (..., 2), not {tensor.shape}")
+        points = tensor.as_subclass(cls)
+        points.canvas_size = canvas_size
+        return points
+
+    if TYPE_CHECKING:
+        # EVIL: Just so that MYPY+PYLANCE+others stop shouting that everything is wrong when initializeing the TVTensor
+        # Not read or defined at Runtime (only at linting time).
+        # TODO: BOUNDING BOXES needs something similar
+        def __init__(
+            self,
+            data: Any,
+            *,
+            canvas_size: Tuple[int, int],
+            dtype: Optional[torch.dtype] = None,
+            device: Optional[Union[torch.device, str, int]] = None,
+            requires_grad: Optional[bool] = None,
+        ):
+            ...
+
+    @classmethod
+    def _wrap_output(
+        cls,
+        output: Any,
+        args: Sequence[Any] = (),
+        kwargs: Optional[Mapping[str, Any]] = None,
+    ) -> Any:
+        # Mostly copied over from the BoundingBoxes TVTensor, minor improvements.
+        # This copies over the metadata.
+        # For BoundingBoxes, that included format, but we only support one format here !
+        flat_params, _ = tree_flatten(args + (tuple(kwargs.values()) if kwargs else ()))  # type: ignore[operator]
+        first_bbox_from_args = next(x for x in flat_params if isinstance(x, KeyPoints))
+        canvas_size: Tuple[int, int] = first_bbox_from_args.canvas_size
+
+        if isinstance(output, torch.Tensor) and not isinstance(output, KeyPoints):
+            output = KeyPoints(output, canvas_size=canvas_size)
+        elif isinstance(output, MutableSequence):
+            # For lists and list-like object we don't try to create a new object, we just set the values in the list
+            # This allows us to conserve the type of complex list-like object that may not follow the initialization API of lists
+            for i, part in enumerate(output):
+                output[i] = KeyPoints(part, canvas_size=canvas_size)
+        elif isinstance(output, Sequence):
+            # Non-mutable sequences handled here (like tuples)
+            # Every sequence that is not a mutable sequence is a non-mutable sequence
+            # We have to use a tuple here, since we know its initialization api, unlike for `output`
+            output = tuple(KeyPoints(part, canvas_size=canvas_size) for part in output)
+        return output
```

Comment:
The purpose of `_wrap_output` is to return a `KeyPoints` object, it's part of a broader interface where we expect the output to be a TVTensor. So we'll need some small changes to this - it also relates to my point above about having a single KeyPoints instance (in the code above there are many)